### ISSUE TITLE
Variable width of columns

### ISSUE THREAD

**Issue by dbeyer** [sosy-lab/benchexec#492]
The columns seem to have all the same width.
The old tables had the width of the columns adjusted to the content.
That is, columns with numbers were taking only a tiny bit of horizontal space.

Horizontal scrolling is currently slowing me down.
There is a lot of unnecessary whitespace.

**Comment by PhilippWendler** [comment]
Yes, that is the downside of the new kind of table. Previously the browser would render everything cell and adjust width, but we cannot use this anymore.

So the only possible option would be to render every cell off screen, measure the width, and apply the width to the column. That would be difficult to implement and considerably slow down loading. Furthermore, in many cases the content of the visible cells changes, e.g., when changing the sort order. Either we precompute the width such that every possible value (even the currently invisible ones) fits, which would mean that we have to prerender *every value* of the whole table, or the widths would have to change on events like resorting, which would be quite confusing.

So basically this feature is not really possible.

What we could do is to adjust the column size based on some heuristic, like type of column (status vs. text vs. numeric), but I am not sure how the heuristic should look like.

**Comment by dbeyer** [comment]
Wow.  This is really a strong downside.

**Comment by PhilippWendler** [comment]
Well, this is the one reason why everything is so much faster now: not every cell has to be rendered immediately.

**Comment by dbeyer** [comment]
Ah, ok.

Then I propose the following idea: Provide an option in the table definition and the user can set the column width to meaningful values. I anyway set the number of digits there in order to control the visual appearance.

**Comment by PhilippWendler** [comment]
I am planning a heuristic that at least adjusts the relative size of the columns according to an estimate of their width.

**Comment by dbeyer** [comment]
This would be really great! ETA?

**Comment by PhilippWendler** [comment]
Please try out the heuristic and check if it works well enough for your tables. Of course, it is not perfect and some superfluous whitespace might be there. But users can still resize columns if it is important.

**Comment by PhilippWendler** [comment]
In the tables I have seen the heuristic worked as expected.
